### DISCUSSION TITLE
tuya OTA bin files

### DISCUSSION THREAD

**Discussion by MattWestb** [zigpy/zigpy#1053] [ZHA development]
Many users is having problem with tuya plugs (TS011F) with one new chip (TElink) that is making stagger things.
One distribute have making one updated firmware and its working for there devices updating thru tuya ZBGW.

One creative (Z2M) user have sniffing the tuya ZBGW and getting the URL to the updated firmware and i adding it in attachment.
[1662545193-oem_zg_tl8258_plug_OTA_3.0.0.zip](https://github.com/zigpy/zigpy/files/9683633/1662545193-oem_zg_tl8258_plug_OTA_3.0.0.zip)

I have looking in it with one hex editor but i dont knowing if its signed or protected in some way. Also i dont knowing if its working in ZHA updating devices with it but i think it shall working.

Now i have one (wrong 2) @puddly question !!!

- 1 Is the bin file standard OTA format and ZHA can using it ?
- 2 It protected / signed or is it possible manipulating the file like the OTA image number ?

Thanks in advance and great thanks for the support for our system !!!

**Comment by Hedda** [comment]
If you get it working then maybe submit the OTA firmware image to Koenkk's zigbee-OTA repo?

https://github.com/Koenkk/zigbee-OTA

https://github.com/Koenkk/zigbee-OTA/tree/master/images/Tuya

**Reply by MattWestb** [reply]
Z2M have adding it and its looks working OK for them.
Im interested if its working in ZHA / Zigpy then its have one other format then the "normal" OTA files.
Also our Puddly is great digging if its signed / protected so if its possible changing the firmware image ID number that i cant telling.

**Comment by remon1496** [comment]
Could you look into creating .ota's of these .bin's also?

https://github.com/Andrik45719/ZY-M100
